dev_tools/oneapi/add_host_function.sh

```diff
@@ -0,0 +1,305 @@
+        if [[ "${temp}" =~ ${config_regex} ]]; then
+            max_num=${rev_id_list[Config]}
+            default_config="${BASH_REMATCH[2]}"
+            temp=""
+            continue
+        fi
+        if [ -n "${template_rm_config}" ]; then
+                template_rm_config="${template_rm_config}, "
+        fi
+        template_rm_config="${template_rm_config}${temp}"
+        if [[  "${temp}" =~ $param_regex ]]; then
+            identification=${BASH_REMATCH[1]}
+            parameter_name=${BASH_REMATCH[2]}
+            # echo "${identification} ${parameter_name}"
+            if [ "${max_num}" -eq "${rev_id_list[Config]}" ]; then
+                if [ "${identification}" != "typename" ]; then
+                    echo "static_assert(false, \"Only inferred type are allowed after Config\");"
+                fi
+            else
+                if [ "${rev_id_list[${identification}]}" -ge "${max_num}" ]; then
+                    max_num=${rev_id_list[${identification}]}
+                    if [ -n "${list_list[${identification}]}" ]; then
+                        parameter_name=", ${parameter_name}"
+                    fi
+                    list_list[${identification}]="${list_list[${identification}]}${parameter_name}"
+                else
+                    echo "static_assert(false, \"Need to follow bool, int, size_type, type ordering\");"
+                fi
+            fi
+        fi
+        temp=""
+    done
+    # for item in "${!list_list[@]}"; do
+    #     echo "//$item - ${list_list[$item]}"
+    # done
+    config_generation=""
+    if [[ "${default_config}" = *"config_set"* ]]; then
+        # When the config is set in template, use it directly.
+        config_generation="::gko::syn::value_list<Config, ${default_config}>();"
+    else
+        # When config use the constexpr variable, use the variable name. The constexpr variable will be changed to Config_list afterwards.
+        config_generation="${default_config}_list"
+    fi
+    echo "template <${template_rm_config}>
+        void ${name}${CONFIG_SELECTION_SUFFIX}
+        (dim3 grid, dim3 block, size_t dynamic_shared_memory, cudaStream_t stream, std::shared_ptr<const CudaExecutor> exec, ${variable}) {
+        auto config_list = ${config_generation};
+        auto exec_config = exec->get_config();
+        ${name}${CONFIG_SELECTION_SUFFIX}(
+            config_list,
+            [&exec_config] (Config config) {
+                return (get_warp_size(exec_config) == get_warp_size(config)) &&
+                        (get_block_size(exec_config) >= get_block_size(config));
+                },
+            ::gko::syn::value_list<${list_list[bool]}>(),
+            ::gko::syn::value_list<${list_list[int]}>(),
+            ::gko::syn::value_list<${list_list[size_type]}>(),
+            ::gko::syn::type_list<${list_list[typename]}>(),
+            grid, block, dynamic_shared_memory, stream, ${var_input});
+    }
+    "
+}
```

Comment:
In general, is this function still used? You have macros for defining this now, no?